### PR TITLE
[Actors] Permit non-sendable, escaping closures to be actor-isolated.

### PR DESCRIPTION
**Pull request by DougGregor**
Remove the heuristic that escaping closures cannot be actor-isolated.
This is in line with the long term plan for actor isolation, but opens
up some holes in the short term because Sendable closures are not enforced
throughout the Swift ecosystem.

One significant effect of this change is that we will now, statically,
fail to detect many cases where actor isolation can be broken.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test

File: test/Concurrency/closure_isolation.swift [hunk 1]
```diff
@@ -22,17 +33,17 @@ extension MyActor {
...
-    // CHECK-NOT: actor-isolated
-    acceptEscapingAsyncClosure { await self.method() }
+    // CHECK: actor-isolated
+    acceptEscapingAsyncClosure { self.syncMethod() }
 
     // CHECK: acceptEscapingAsyncClosure
     // CHECK: closure_expr
-    // CHECK-NOT:actor-isolated
-    acceptEscapingAsyncClosure { () async in print("hello") }
+    // CHEC:actor-isolated
```
Comment (ktoso) [inline test/Concurrency/closure_isolation.swift:45]: typo

```suggestion
    // CHECK:actor-isolated
```

**Review by ktoso — APPROVED** [review]
Unfortunate without Sendable checking but LGTM.

You have one typo in the tests which means it won't check one of the properties :-)

**Comment by DougGregor** [comment]
@swift-ci please smoke test macOS

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test Linux

File: test/Concurrency/closure_isolation.swift (hunk 1 above)
Comment (DougGregor) [reply test/Concurrency/closure_isolation.swift:45]: Whoops, thanks!

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test macOS

**Pull request merged**